moz_sql_parser/formatting.py

```diff
@@ -128,6 +128,11 @@ def op(self, json):
             return self._on(json)
 
         if len(json) > 1:
+            # Nested queries
+            if 'from' in json:
+                return '({})'.format(self.format(json))
+            if 'select' in json:
+                return '({})'.format(self.format(json))
```

Comment:
Move these two checks to the `dispatch` function: It is responsible for probing the json structure to determine what formatter to use.

Reply:
Moved. Please check the new commit.